### PR TITLE
Add deprecation warnings for API changes

### PR DESCRIPTION
**Pull request by jasongrout**
This is a followup of #979, correcting our LegendControl docs where needed.

I left in a backwards-compatible shim for giving the name argument to the LegendControl, since that is likely the most common thing people would have to change.

### CODE REVIEW

**Comment by jasongrout** [comment]
Also I realized #982 wasn't documented in the changelog, so I did that too.

**Comment by jasongrout** [comment]
I created an 0.17.0 milestone to have the issues I think should be in the release. If those two issues pass review and are merged, I am happy to do the 0.17.0 release:

https://github.com/jupyter-widgets/ipyleaflet/milestone/1

**Comment by jasongrout** [comment]
@HaudinFlorence, you could also review this. Thanks again for the changes cleaning the api up!

File: ipyleaflet/leaflet.py [hunk 1]
```diff
@@ -1932,10 +1932,13 @@ class LegendControl(Control):
         "value 2": "#55A",
         "value 3": "#005"}).tag(sync=True)
 
-    def __init__(self, legend, *args, name="Legend", **kwargs):
+    def __init__(self, legend, *args, **kwargs):
+        kwargs["legend"] = legend
+        # For backwards compatibility with ipyleaflet<=0.16.0
+        if 'name' in kwargs:
+            kwargs.setdefault('title', kwargs['name'])
```
Comment (martinRenou) [inline ipyleaflet/leaflet.py:1939]: Should we show a warning in this case?

Comment (jasongrout) [reply ipyleaflet/leaflet.py:1939]: We could, but I'm also happy to support it as an undocumented option for now for backwards compatibility. We might ask ourselves why we don't make the other changes just deprecations at this point too, and remove them at some later point, to ease backwards compatibility.

Do you want it deprecated?

Comment (jasongrout) [reply ipyleaflet/leaflet.py:1944]: I added a bunch of deprecation warnings, and added back the removed apis to be deprecated. I also changed the pendingdeprecationwarnings to just deprecation warnings.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,13 +1,32 @@
...
 ### New Features
 
 * Make it possible to use Choropleth layer with data containing NaNs [#972](https://github.com/jupyter-widgets/ipyleaflet/pull/972)
 * Add Map panes [#999](https://github.com/jupyter-widgets/ipyleaflet/pull/999)
 * Allow setting Map.dragging [#1001](https://github.com/jupyter-widgets/ipyleaflet/pull/1001)
 * Add visible attribute to GeoJSON layer [#1002](https://github.com/jupyter-widgets/ipyleaflet/pull/1002)
-* [BREAKING CHANGE] Remove get and set decorators in LegendControl [#979](https://github.com/jupyter-widgets/ipyleaflet/pull/979)
 
-## Maintenance
+### Breaking Changes
```
Comment (jasongrout) [inline CHANGELOG.md:12]:
```suggestion
### Deprecated API
```

**Review by martinRenou — APPROVED** [review]
Thanks!

**Pull request merged**